jetty-server/src/main/java/org/eclipse/jetty/server/HttpOutput.java

```diff
@@ -53,16 +53,21 @@
  */
 public class HttpOutput extends ServletOutputStream implements Runnable
 {
-    private static Logger LOG = Log.getLogger(HttpOutput.class);
-    private final HttpChannel<?> _channel;
-    private final SharedBlockingCallback _writeblock=new SharedBlockingCallback()
+
+    public interface Interceptor
```

Comment:
We don't actually need or want the interceptor interface.   I think we can get by just with the `isOptimizedForDirectBuffers` method directly added to the `HttpChannel` class without an `@Override`, then `_interceptor.isOptimizedForDirectBuffers()` can be replaced with `_channel.isOptimizedForDirectBuffers()`